pkg/virt-controller/watch/vmi.go

```diff
@@ -1326,6 +1319,33 @@ func (c *VMIController) virtlauncherAttachmentPods(virtlauncherPod *k8sv1.Pod) (
 	return attachmentPods, nil
 }
 
+func (c *VMIController) waitForFirstConsumerTemporaryPods(vmi *virtv1.VirtualMachineInstance, virtLauncherPod *k8sv1.Pod) ([]*k8sv1.Pod, error) {
+	var temporaryPods []*k8sv1.Pod
+
+	// Get all pods from the namespace
+	pods, err := c.listPodsFromNamespace(vmi.Namespace)
+	if err != nil {
+		return temporaryPods, err
+	}
+
+	for _, pod := range pods {
+		// Cleanup candidates are temporary pods that are either controlled by the VMI or the virt launcher pod
+		if !isTempPod(pod) {
+			continue
+		}
+
+		if controller.IsControlledBy(pod, vmi) {
```

Comment:
Would the following work instead?

```
if controller.IsControlledBy(pod, vmi)  || controller.IsControlledBy(pod, virtLauncherPod) {
```

Reply:
Then you could delete 1341-1344?

Reply:
This won't work since `controller.IsControlledBy` (which is what we use) is strictly for a Pod&VMI combo:
https://github.com/kubevirt/kubevirt/blob/main/pkg/controller/controller_ref.go#L45
However, there is  https://github.com/kubevirt/kubevirt/blob/main/vendor/k8s.io/apimachinery/pkg/apis/meta/v1/controller_ref.go#L24 that we do not use, and I'm not sure why